pkg/controllers/configobservation/configobservercontroller/apply_configuration_test.go

```diff
@@ -0,0 +1,140 @@
+package configobservercontroller
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	applyoperatorv1 "github.com/openshift/client-go/operator/applyconfigurations/operator/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	configinformer "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/manifestclient"
+	libgoetcd "github.com/openshift/library-go/pkg/operator/configobserver/etcd"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+func TestApplyConfiguration(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	mutationTrackingHTTPClient := manifestclient.NewHTTPClient("/home/deads/Downloads/must-gather-01")
```

Comment:
does it mean we will produce (the input dir) a directory structure similar to a must-gather?

Reply:
> does it mean we will produce (the input dir) a directory structure similar to a must-gather?

Yes. It's an interchange format we have tooling for already.